qt/app.py

```diff
@@ -1211,12 +1211,27 @@ def updateTimeLine(self, refreshSnapshotsList=True):
                 item = self.timeLine.addSnapshot(sid)
             self.timeLine.checkSelection()
 
+    def validate_on_take_snapshot(self):
+        missing = snapshots.has_missing(self.config.include())
+        if missing:
+            msg_missing = '\n'.join(missing)
+            msg = _('The following folders are missing: {folders} Do you want to proceed?'.format(
```

Comment:
The `def _warning_on_take_snapshot()` in snapshots.py also has a similar message creation logic so this code partially looks like duplicated code (and "includes" could also be files so I would add "**files**/folders` here too.

If you'd find a simple way to de-duplicate the code it would be great but if it is too complicated I could live with the existing code here.

Reply:
About this, a similar code was made with the `missing = snapshots.has_missing(self.config.include())` (which is now renamed to `has_missing_includes`)

Only the text generation is different (also not translated the CLI version) and how we warn the user.

Ok?